src/components/Unicon.vue

```diff
@@ -66,17 +78,30 @@ export default {
       if (icon) {
         return icon.path
       } else {
-        console.error(`Name '${ this.name }' of the icon is not correct`)
+        console.error(`Name '${this.name}' of the icon is not correct`)
         return undefined
       }
     }
+  },
+  methods: {
+    onHover () {
+      if (this.hoverFill) {
+        this.localFill = this.hoverFill
+      }
+    },
+    onLeave () {
+      if (this.hoverFill) {
+        this.localFill = this.fill
+      }
+    }
   }
 }
 </script>
 
 <style>
 .unicon {
   display: inline-block;
+  transition: 0.3s all;
```

Comment:
Let's reduce it to 0.2.s, it's too long ). And by the way, as it turns out `fill: var(--uni-color-primary);` is not used, so you can remove that. To add a transition, you need to add it like this - `.unicon svg {transition: 0.2s all}`